### PR TITLE
feat!: allow custom seed when using targetingKey override for fractional op

### PR DESCRIPTION
**Pull request by colebaileygit**
## This PR

- no longer injects flagKey as seed for fractional op when user has provided custom targeting
- updates schema to allow `cat` and other operations so that custom targeting can be properly seeded

### Related Issues

https://github.com/open-feature/flagd/issues/1264

### How to test

```bash
# unit tests
make test

# gherkin tests
./bin/flagd start -f file:test-harness/flags/testing-flags.json -f file:test-harness/flags/custom-ops.json -f file:test-harness/flags/evaluator-refs.json -f file:test-harness/flags/zero-flags.json
make flagd-integration-test
```

### CODE REVIEW

File: core/pkg/evaluator/fractional.go [hunk 1]
```diff
@@ -114,6 +116,8 @@ func distributeValue(value string, feDistribution []fractionalEvaluationDistribu
 	hashRatio := math.Abs(float64(hashValue)) / math.MaxInt32
 	bucket := int(hashRatio * 100) // in range [0, 100]
 
+	fmt.Println(value, feDistribution)
```
Comment (Kavindu-Dodan) [inline core/pkg/evaluator/fractional.go:119]:
```suggestion
```

no prints ;)

**Comment by Kavindu-Dodan** [comment]
@colebaileygit could you please update the documentation [1]. We should document the ability to customize the distribution key (ex:- using cat or any other valid string operation). And include not on this change.

[1] - https://github.com/open-feature/flagd/blob/main/docs/reference/custom-operations/fractional-operation.md

File: core/pkg/evaluator/fractional.go (hunk 1 above)
Comment (colebaileygit) [reply core/pkg/evaluator/fractional.go:119]: oops, thank you!

**Comment by colebaileygit** [comment]
> @colebaileygit could you please update the documentation [1]. We should document the ability to customize the distribution key (ex:- using cat or any other valid string operation). And include not on this change.
> 
> [1] - https://github.com/open-feature/flagd/blob/main/docs/reference/custom-operations/fractional-operation.md
Added 👍

File: docs/reference/custom-operations/fractional-operation.md [hunk 2]
```diff
@@ -46,8 +52,9 @@ is selected.
 As hashing is deterministic we can be sure to get the same result every time for the same data point.
 
 The `fractional` operation can be added as part of a targeting definition.
-The value is an array and the first element is the name of the property to use from the evaluation context.
+The value is an array and the first element is a nested JsonLogic rule which resolves to the hash key. This rule should typically consist of a seed concatenated with a session variable to use from the evaluation context.
```
Comment (beeme1mr) [inline docs/reference/custom-operations/fractional-operation.md:55]:
```suggestion
The value is an array and the first element is a nested JsonLogic rule which resolves to the hash key.
This rule should typically consist of a seed concatenated with a session variable to use from the evaluation context.
```

It's a bit easy to review when each sentence is on its own line.

**Review by beeme1mr — APPROVED** [review]

**Review by Kavindu-Dodan — APPROVED** [review]

**Comment by Kavindu-Dodan** [comment]
@colebaileygit I have updated this PR with the latest submodules as well as Go dependencies. All checks are green. Planning to merge this tomorrow :)

**Review by toddbaert — APPROVED** [review]

**Comment by toddbaert** [comment]
Great solution and effort @colebaileygit . Thanks for your diligence.

@Kavindu-Dodan @beeme1mr I guess we need to update the in-process impls ASAP?

**Comment by Kavindu-Dodan** [comment]
> Great solution and effort @colebaileygit . Thanks for your diligence.
> 
> @Kavindu-Dodan @beeme1mr I guess we need to update the in-process impls ASAP?

Yeah, for Go SDK, it's a matter of using the latest flagd core with the next release.

Otherwise, we already have PRs for Java [1] & Python [2]. I think JS needs to be taken care of. 

[1] - https://github.com/open-feature/java-sdk-contrib/pull/737
[2] - https://github.com/open-feature/python-sdk-contrib/pull/74

**Pull request merged**